### PR TITLE
feature: Add Neuronx Image uri - Transformers 4.28 - PyTorch 1.13

### PR DESCRIPTION
**Pull request by JingyaHuang**
*Issue #3845:*

*Description of changes:*

Add Image URI for:
* [Transformers 4.28 - PyTorch 1.13.0 - Neuronx Training DLC](https://github.com/aws/deep-learning-containers/releases/tag/v1.0-hf-4.28.1-pt-1.13.0-tr-neuronx-sdk2.9.1-py38)
* [Transformers 4.28 - PyTorch 1.13.0 - Neuronx Inference DLC](https://github.com/aws/deep-learning-containers/releases/tag/v1.0-hf-4.28.1-pt-1.13.0-inf-neuronx-sdk2.9.1-py38)

*Testing done:*

## Merge Checklist

_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your pull request._

#### General

- [ ] I have read the [CONTRIBUTING](https://github.com/aws/sagemaker-python-sdk/blob/master/CONTRIBUTING.md) doc
- [ ] I certify that the changes I am introducing will be backward compatible, and I have discussed concerns about this, if any, with the Python SDK team
- [ ] I used the commit message format described in [CONTRIBUTING](https://github.com/aws/sagemaker-python-sdk/blob/master/CONTRIBUTING.md#committing-your-change)
- [ ] I have passed the region in to all S3 and STS clients that I've initialized as part of this change.
- [ ] I have updated any necessary documentation, including [READMEs](https://github.com/aws/sagemaker-python-sdk/blob/master/README.rst) and [API docs](https://github.com/aws/sagemaker-python-sdk/tree/master/doc) (if appropriate)

#### Tests

- [ ] I have added tests that prove my fix is effective or that my feature works (if appropriate)
- [ ] I have added unit and/or integration tests as appropriate to ensure backward compatibility of the changes
- [ ] I have checked that my tests are not configured for a specific region or account (if appropriate)
- [ ] I have used [`unique_name_from_base`](https://github.com/aws/sagemaker-python-sdk/blob/master/src/sagemaker/utils.py#L77) to create resource names in integ tests (if appropriate)

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Review by mufaddal-rohawala — APPROVED** [review]

**Review by mufaddal-rohawala — COMMENTED** [review]
/bot run all

**Comment by mufaddal-rohawala** [comment]
@JingyaHuang can you fix unit tests, by locally running - `tox -e black-format`

**Review by knikure — COMMENTED** [review]
/bot run slow-tests

**Comment by knikure** [comment]
@JingyaHuang Can you fix the black-check errors by running `tox -e black-format` command?
Submit your changes after successfully running `./.githooks/pre-push` command.

**Review by trajanikant — COMMENTED** [review]
/bot run all

**Comment by JingyaHuang** [comment]
Hi folks @trajanikant @knikurem, thanks for reviewing. I did some tentative modifications to the tests, usually we only update image uri. Shall I remove those changes?

**Review by mufaddal-rohawala — APPROVED** [review]

**Review by mufaddal-rohawala — COMMENTED** [review]
/bot run all

**Comment by mufaddal-rohawala** [comment]
> 

@JingyaHuang thanks for adding those tests, we need those. Can you please fix the unit tests to quickly unblock this PR. Just run `tox -e black-format` and commit those changes.

**Comment by JingyaHuang** [comment]
@mufaddal-rohawala Thanks, I just fixed the style.

**Comment by mufaddal-rohawala** [comment]
@JingyaHuang This time some unit tests are failing, please have a look at the failed logs link in the sagemaker-bot comments above. Slow tests seems to be unrelated to the change, Il retry those.

**Review by mufaddal-rohawala — COMMENTED** [review]
/bot run slow-tests

**Review by mufaddal-rohawala — COMMENTED** [review]
/bot run slow-tests

**Comment by claytonparnell** [comment]
@JingyaHuang Hello, the unit test is failing because `sagemaker/image_uris::_get_inference_tool` is currently returning just `neuron`, rather than `neuronx`, so the config for `huggingface-neuron` is incorrectly getting fetched.

**Review by claytonparnell — COMMENTED** [review]
/bot run all

File: src/sagemaker/huggingface/model.py [hunk 1]
```diff
@@ -534,4 +544,5 @@ def serving_image_uri(
             image_scope="inference",
             base_framework_version=base_framework_version,
             serverless_inference_config=serverless_inference_config,
+            inference_tool=inference_tool,
```
Comment (mufaddal-rohawala) [inline src/sagemaker/huggingface/model.py:547]: This sounds reasonable to me, how did this work for `neuron` before.

Comment (claytonparnell) [reply src/sagemaker/huggingface/model.py:547]: If no inference_tool, there's a function `sagemaker/image_uris::_get_inference_tool` which returns 'neuron' `if instance_type_family.startswith("inf") or instance_type_family.startswith("trn")`

Comment (JingyaHuang) [reply src/sagemaker/huggingface/model.py:547]: @claytonparnell Thanks for helping on debugging the tests. It looks good from my side.

**Pull request merged**